client/java/generator/src/main/java/io/openlineage/client/JavaPoetGenerator.java

```diff
@@ -320,6 +327,10 @@ private TypeSpec builderClass(ObjectResolvedType type) {
       // only regular fields are used in the builder
       @Override
       public void onField(ResolvedField f) {
+        if (f.getConstantValue().isPresent()) {
```

Comment:
instead of adding this if statement in every call to onField(), you could add an onConstantField() to the Visitor with the default implementation that does nothing.